File: src/scripts/download_models.js
const https = require('https');
const fs = require('fs');
const path = require('path');

const MODELS = {
    "epicrealism_naturalSinRC1VAE.safetensors": "https://huggingface.co/Justin-Choo/epiCRealism-Natural_Sin_RC1_VAE/resolve/main/epicrealism_naturalSinRC1VAE.safetensors",
    "ponyDiffusionV6XL_v6StartWithThisOne.safetensors": "https://huggingface.co/Runware/Pony_Diffusion_V6_XL/resolve/main/ponyDiffusionV6XL_v6StartWithThisOne.safetensors"
};

const OUTPUT_DIR = "D:\\ComfyUI_Models\\checkpoints";

function downloadFile(url, dest) {
    return new Promise((resolve, reject) => {
        const file = fs.createWriteStream(dest);
        https.get(url, (response) => {
            if (response.statusCode === 301 || response.statusCode === 302 || response.statusCode === 307) {
                console.log(`Redirecting to: ${response.headers.location}`);
                return downloadFile(response.headers.location, dest).then(resolve).catch(reject);
            }
            if (response.statusCode !== 200) {
                reject(new Error(`Failed to get '${url}' (${response.statusCode})`));
                return;
            }
            response.pipe(file);
            file.on('finish', () => {
                file.close(resolve);
            });
        }).on('error', (err) => {
            fs.unlink(dest, () => {});
            reject(err);
        });
    });
}

async function runDownloads() {
    if (!fs.existsSync(OUTPUT_DIR)) {
        fs.mkdirSync(OUTPUT_DIR, { recursive: true });
    }
    
    for (const [filename, url] of Object.entries(MODELS)) {
        const dest = path.join(OUTPUT_DIR, filename);
        if (fs.existsSync(dest) && fs.statSync(dest).size > 1000000) {
            console.log(`[+] ${filename} already exists, skipping.`);
            continue;
        }
        console.log(`[ ] Downloading ${filename}...`);
        try {
            await downloadFile(url, dest);
            console.log(`[+] Successfully downloaded ${filename}`);
        } catch (err) {
            console.error(`[-] Failed to download ${filename}: ${err.message}`);
        }
    }
    console.log("All downloads completed.");
}

runDownloads();

File: src/scripts/download_models.py
import urllib.request
import os
import ssl

ssl._create_default_https_context = ssl._create_unverified_context

MODELS = {
    "Juggernaut-XL_v9_RunDiffusionPhoto_v2.safetensors": "https://huggingface.co/RunDiffusion/Juggernaut-XL-v9/resolve/main/Juggernaut-XL_v9_RunDiffusionPhoto_v2.safetensors",
    "epiCRealism_naturalSinRC1VAE.safetensors": "https://huggingface.co/emilianJR/epiCRealism/resolve/main/epCRealism_naturalSinRC1VAE.safetensors",
    "ponyDiffusionV6XL_v6StartWithThisOne.safetensors": "https://huggingface.co/AstraliteHeart/pony-diffusion-v6-xl/resolve/main/ponyDiffusionV6XL_v6StartWithThisOne.safetensors"
}

OUTPUT_DIR = r"I:\ComfyUI_windows_portable\ComfyUI\models\checkpoints"

def download_model(filename, url):
    output_path = os.path.join(OUTPUT_DIR, filename)
    if os.path.exists(output_path):
        print(f"{filename} already exists, skipping.")
        return
        
    print(f"Downloading {filename}...")
    try:
        urllib.request.urlretrieve(url, output_path)
        print(f"Successfully downloaded {filename}")
    except Exception as e:
        print(f"Failed to download {filename}: {e}")

if __name__ == "__main__":
    if not os.path.exists(OUTPUT_DIR):
        print(f"Output directory not found: {OUTPUT_DIR}")
    else:
        for filename, url in MODELS.items():
            download_model(filename, url)
        print("All downloads completed.")
